gqcp/include/Operator/SecondQuantized/USQTwoElectronOperator.hpp

```diff
@@ -0,0 +1,511 @@
+     */
+    template <size_t Z = Components>
+    USQTwoElectronOperator(const QCRankFourTensor<Scalar>& g_alphaAlpha, const QCRankFourTensor<Scalar>& g_alphaBeta, const QCRankFourTensor<Scalar>& g_betaAlpha, const QCRankFourTensor<Scalar>& g_betaBeta, typename std::enable_if<Z == 1>::type* = 0) :
+        USQTwoElectronOperator(std::array<QCRankFourTensor<Scalar>, 1>{g_alphaAlpha}, std::array<QCRankFourTensor<Scalar>, 1>{g_alphaBeta}, std::array<QCRankFourTensor<Scalar>, 1>{g_betaAlpha}, std::array<QCRankFourTensor<Scalar>, 1>{g_betaBeta})
+    {}
+
+
+    /**
+     *  Construct an unrestricted two-electron operator with zero parameters, dimensions of alpha and beta component are the same.
+     * 
+     *  @param dim        the dimension of the matrix representation of the alpha and beta parameters, i.e. the number of orbitals/sites
+     * 
+     */
+    USQTwoElectronOperator(const size_t dim) {
+        for (size_t i = 0; i < Components; i++) {
+            this->gs_alphaAlpha[i] = QCRankFourTensor<Scalar>(dim);
+            this->gs_alphaBeta[i] = QCRankFourTensor<Scalar>(dim);
+            this->gs_betaAlpha[i] = QCRankFourTensor<Scalar>(dim);
+            this->gs_betaBeta[i] = QCRankFourTensor<Scalar>(dim);
+            this->gs_alphaAlpha[i].setZero();
+            this->gs_alphaBeta[i].setZero();
+            this->gs_betaAlpha[i].setZero();
+            this->gs_betaBeta[i].setZero();
+        }
+    }
+
+    
+    /**
+     *  Default constructor: construct an unrestricted two-electron operator with parameters that are zero
+     */
+    USQTwoElectronOperator() :
+        USQTwoElectronOperator(0)  // dimensions of the representations are zero
+    {}
+
+
+    /*
+     *  PUBLIC METHODS
+     */
+
+    /**
+     *  @return read-only tensor representations of all the alpha-alpha parameters (integrals) of the different components of this second-quantized operator
+     */
+    const std::array<QCRankFourTensor<Scalar>, Components>& allAlphaAlphaParameters() const {
+        return this->gs_alphaAlpha;
+    }
+
+    /**
+     *  @return writable tensor representations of all the alpha-alpha parameters (integrals) of the different components of this second-quantized operator
+     */
+    std::array<QCRankFourTensor<Scalar>, Components>& allAlphaAlphaParameters() {
+        return this->gs_alphaAlpha;
+    }
+
+
+    /**
+     *  @return read-only tensor representations of all the alpha-beta parameters (integrals) of the different components of this second-quantized operator
+     */
+    const std::array<QCRankFourTensor<Scalar>, Components>& allAlphaBetaParameters() const {
+        return this->gs_alphaBeta;
+    }
+
+    /**
+     *  @return writable tensor representations of all the alpha-beta parameters (integrals) of the different components of this second-quantized operator
+     */
+    std::array<QCRankFourTensor<Scalar>, Components>& allAlphaBetaParameters() {
+        return this->gs_alphaBeta;
+    }
```

Comment:
As you can see, there's a lot of copy-pasting going on here. Could you rewrite the accessor methods, using the `GQCP::SpinComponent` enum.

You could use the following API, for example:
```cpp
const std::array<QCRankFourTensor<Scalar>, Components>& allParameters(SpinComponent, SpinComponent) const;

std::array<QCRankFourTensor<Scalar>, Components>& allParameters(SpinComponent, SpinComponent);
```

Reply:
I didn't think of that, and it is indeed a way cleaner method to write these methods. Thanks for the advice!